app/assets/main/react_components/dashboard/climate_actions/carousel/CarouselCategoryButton.jsx

```diff
@@ -31,26 +36,29 @@ const CarouselCategoryButton = ({
   categories && createOptions();
 
   const handleCategory = (categoryID) => {
-    categoryID != null ? categoryClick(categoryID) : allCategoriesClick();
+    let filteredActions = [];
+    if (categoryID === "popular") {
+      categoryClick(true, false);
+      filteredActions = actionsToplist.slice(0, 5);
+    } else if (categoryID === "allCategories") {
+      categoryClick(false, true);
+      filteredActions = totClimateActions;
+    } else {
+      categoryClick(false, false, categoryID);
+      filteredActions = totClimateActions.filter(
+        (temp) => temp.climate_action_category_id === categoryID
+      );
+    }
+    updateCategoryAttributes(filteredActions);
   };
 
-  const allCategoriesClick = () => {
-    setAllCategories(true);
-    updateCategory();
+  const categoryClick = (popularValue, allCategoriesValue, categoryID) => {
+    setPopular(popularValue);
```

Comment:
If this is a boolean we should name it something like "showPopular" or "popularClicked"

Reply:
Same with allCategoriesValue